### PR TITLE
add r-forestfloor

### PR DESCRIPTION
**Pull request by fhoehle**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by fhoehle** [comment]
@conda-forge/staged-recipes This is ready for review/merge.

File: recipes/r-forestfloor/build.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+if [[ $target_platform =~ linux.* ]] || [[ $target_platform == win-32 ]] || [[ $target_platform == win-64 ]] || [[ $target_platform == osx-64 ]]; then
+  export DISABLE_AUTOBREW=1
+  X11_CONFIGURE_ARGS="--x-includes=${PREFIX}/include --x-libraries=${BUILD_PREFIX}/x86_64-conda_cos6-linux-gnu/sysroot/usr/lib64"
+  export LD_LIBRARY_PATH=${BUILD_PREFIX}/x86_64-conda_cos6-linux-gnu/sysroot/usr/lib64
```
Comment (isuruf) [inline recipes/r-forestfloor/build.sh]: Why is this necessary?

Comment (fhoehle) [reply recipes/r-forestfloor/build.sh]: I stole it from [r-rgl](https://github.com/conda-forge/r-rgl-feedstock/blob/master/recipe/build.sh). Without building fails with missing `libGL.so.1` error. If you know a better solution I'm happy see.

Comment (fhoehle) [reply recipes/r-forestfloor/build.sh]: @isuruf I shamelessly stole it from [r-rgl](https://github.com/conda-forge/r-rgl-feedstock/blob/master/recipe/build.sh). Without building fails with missing libGL.so.1 error. If you know a better solution I'm happy see.

Comment (isuruf) [reply recipes/r-forestfloor/build.sh]: Do you have a log?

Comment (fhoehle) [reply recipes/r-forestfloor/build.sh]: @isuruf here the relevant part
```
...
** package ‘forestFloor’ successfully unpacked and MD5 sums checked
** libs
installing to /opt/conda/conda-bld/r-forestfloor_1563684818758/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib/R/library/forestFloor/libs
** R
** inst
** byte-compile and prepare package for lazy loading
Error in dyn.load(file, DLLpath = DLLpath, ...) : 
  unable to load shared object '/opt/conda/conda-bld/r-forestfloor_1563684818758/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib/R/library/rgl/libs/rgl.so':
  libGL.so.1: cannot open shared object file: No such file or directory
Error : .onLoad failed in loadNamespace() for 'rgl', details:
  call: NULL
  error:        Loading rgl's DLL failed. 
ERROR: lazy loading failed for package ‘forestFloor’
* removing ‘/opt/conda/conda-bld/r-forestfloor_1563684818758/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib/R/library/forestFloor’
-GCC_RANLIB=$PREFIX/bin/x86_64-conda_cos6-linux-gnu-gcc-ranlib
INFO: activate-gxx_linux-64.sh made the following environmental changes:
+CXX=$BUILD_PREFIX/bin/x86_64-conda_cos6-linux-gnu-c++
-CXX=$PREFIX/bin/x86_64-conda_cos6-linux-gnu-c++
+GXX=$BUILD_PREFIX/bin/x86_64-conda_cos6-linux-gnu-g++
-GXX=$PREFIX/bin/x86_64-conda_cos6-linux-gnu-g++
x86_64-conda_cos6-linux-gnu-c++  -I"$PREFIX/lib/R/include" -DNDEBUG  -I"$PREFIX/lib/R/library/Rcpp/include" -DNDEBUG -D_FORTIFY_SOURCE=2 -O2  -I$PREFIX/include -Wl,-rpath-link,$PREFIX/lib   -fpic  -fvisibility-inlines-hidden  -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -I$PREFIX/include -fdebug-prefix-map=/home/conda/feedstock_root/build_artifacts/r-base_1562026475051/work=/usr/local/src/conda/r-base-3.5.1 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix  -c RcppExports.cpp -o RcppExports.o
x86_64-conda_cos6-linux-gnu-c++  -I"$PREFIX/lib/R/include" -DNDEBUG  -I"$PREFIX/lib/R/library/Rcpp/include" -DNDEBUG -D_FORTIFY_SOURCE=2 -O2  -I$PREFIX/include -Wl,-rpath-link,$PREFIX/lib   -fpic  -fvisibility-inlines-hidden  -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -I$PREFIX/include -fdebug-prefix-map=/home/conda/feedstock_root/build_artifacts/r-base_1562026475051/work=/usr/local/src/conda/r-base-3.5.1 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix  -c testcpp_rec6.cpp -o testcpp_rec6.o
x86_64-conda_cos6-linux-gnu-c++ -shared -L$PREFIX/lib/R/lib -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags -Wl,--gc-sections -Wl,-rpath,$PREFIX/lib -Wl,-rpath-link,$PREFIX/lib -L$PREFIX/lib -Wl,-rpath-link,$PREFIX/lib -o forestFloor.so RcppExports.o testcpp_rec6.o -L$PREFIX/lib/R/lib -lR
Traceback (most recent call last):
  File "/opt/conda/bin/conda-build", line 11, in <module>
    sys.exit(main())
  File "/opt/conda/lib/python3.7/site-packages/conda_build/cli/main_build.py", line 445, in main
    execute(sys.argv[1:])
  File "/opt/conda/lib/python3.7/site-packages/conda_build/cli/main_build.py", line 436, in execute
    verify=args.verify, variants=args.variants)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/api.py", line 209, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 2324, in build_tree
    notest=notest,
  File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 1472, in build
    cwd=src_dir, stats=build_stats)
```

Comment (isuruf) [reply recipes/r-forestfloor/build.sh]: You need a `yum_requirements.txt` file like https://github.com/conda-forge/fltk-feedstock/blob/fcb778d2fd11859214323ac4bf0361354fef9451/recipe/yum_requirements.txt

Comment (fhoehle) [reply recipes/r-forestfloor/build.sh]: @isuruf Thx! I've adapted accordingly, removing `export LD_LIBRARY_PATH= ...` and added a `yum_requirements.txt`

File: recipes/r-forestfloor/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    - {{ compiler('m2w64_c') }}        # [win]
+    - {{ compiler('cxx') }}            # [not win]
+    - {{ compiler('m2w64_cxx') }}      # [win]
+    - {{posix}}filesystem        # [win]
+    - {{posix}}make
+    - {{posix}}sed               # [win]
+    - {{posix}}coreutils         # [win]
+    - {{posix}}zip               # [win]
+    - {{ cdt('mesa-libgl-devel') }}      # [linux]
+    - {{ cdt('mesa-libgl-devel') }}      # [linux]
```
Comment (cbrueffer) [inline recipes/r-forestfloor/meta.yaml]: Duplicated line

Comment (fhoehle) [reply recipes/r-forestfloor/meta.yaml]: Thx @cbrueffer , removed!

**Comment by fhoehle** [comment]
@isuruf and @cbrueffer appveyor build was successful.

File: recipes/r-forestfloor/build.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+if [[ $target_platform =~ linux.* ]] || [[ $target_platform == win-32 ]] || [[ $target_platform == win-64 ]] || [[ $target_platform == osx-64 ]]; then
+  export DISABLE_AUTOBREW=1
+  X11_CONFIGURE_ARGS="--x-includes=${PREFIX}/include --x-libraries=${BUILD_PREFIX}/x86_64-conda_cos6-linux-gnu/sysroot/usr/lib64"
```
Comment (cbrueffer) [inline recipes/r-forestfloor/build.sh]: Is this and the LD_LIBRARY_PATH in the meta.yaml tests still necessary? I applied the yum_requirements.txt fix to https://github.com/conda-forge/r-oceanview-feedstock/pull/2 (failed with the `libGL not found` error as well) and it built successfully without additional changes.

Comment (FelixHoehleQC) [reply recipes/r-forestfloor/build.sh]: @cbrueffer done!

Comment (fhoehle) [reply recipes/r-forestfloor/build.sh]: @cbrueffer done

File: recipes/r-forestfloor/build.sh [hunk 4]
```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+if [[ $target_platform =~ linux.* ]] || [[ $target_platform == win-32 ]] || [[ $target_platform == win-64 ]] || [[ $target_platform == osx-64 ]]; then
+  export DISABLE_AUTOBREW=1
+
+  DISPLAY=${DISPLAY:-:0} $R CMD INSTALL --build .
```
Comment (cbrueffer) [inline recipes/r-forestfloor/build.sh]: Thanks for the updates! I meant the DISPLAY part here as well, still needed?

Comment (fhoehle) [reply recipes/r-forestfloor/build.sh]: @cbrueffer no :-), is not needed. Removed.

File: recipes/r-forestfloor/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+build:
+  skip: True              # [osx]
+  merge_build_host: True  # [win]
+  number: 0
+  rpaths:
+    - lib/R/lib/
+    - lib/
+  script_env:
+    - DISPLAY
```
Comment (cbrueffer) [inline recipes/r-forestfloor/meta.yaml]: This is also unneeded now I suppose?

Comment (fhoehle) [reply recipes/r-forestfloor/meta.yaml]: correct

Comment (fhoehle) [reply recipes/r-forestfloor/meta.yaml]: @cbrueffer removed `script_env`

**Comment by cbrueffer** [comment]
Thanks @fhoehle !
